panther_analysis_tool/backend/public_api_client.py

```diff
@@ -347,6 +378,28 @@ def _execute(
             request, variable_values=variable_values, get_execution_result=True
         )
 
+    def _safe_execute(
+        self,
+        request: DocumentNode,
+        variable_values: Optional[Dict[str, Any]] = None,
+    ) -> ExecutionResult:
+        try:
+            res = self._execute(request, variable_values=variable_values)
+        except TransportQueryError as e:  # pylint: disable=C0103
+            err = PermanentBackendError(e)
+            if e.errors and len(e.errors) > 0:
+                err = PermanentBackendError(e.errors[0])
+                err.permanent = not is_retryable_error(e.errors[0])
```

Comment:
Does this mean that you can have a PermanentBackendError with permanent == False? That doesn't make sense to me

Reply:
good point, I'll update this one to a BackendError. I added the new `PermanentBackendError` type because the `BackendError` type by default has permanent set to False .. there were some bugs introduced due to that behavior so wanted to be explicit